ngrok/ngrok-reconcile-uiresource.sh

```diff
@@ -0,0 +1,89 @@
+#!/bin/bash
+#
+# Reconcilation for each uiresource.
+#
+# Given the name of a UIResource, we check:
+# 1) Does it have any localhost links?
+# 2) If so, attach an ngrok button that enables/disables the tunnel
+
+set -euo pipefail
+
+name="$1"
+echo "reconciling $name"
+
+disabled="true"
+ports=()
+links="$(tilt get uiresource "$name" -o jsonpath='{range .status.endpointLinks[*]}{.url}{"\n"}{end}')"
+for link in $links;
+do
+    host="localhost"
+    port="$(echo "$link" | sed -e "s|^http://$host:\([0-9]*\).*|\1|")"
+    if [[ "$port" != "$link" ]]; then
+        ports+=("$port")
+        disabled="false"
+    fi
+done
+
+child_name="ngrok:$name"
+if [[ "$disabled" == "true" ]]; then
+    tilt delete uibutton "$child_name" --ignore-not-found
+    tilt delete configmap "$child_name" --ignore-not-found
+    tilt delete cmd "$child_name" --ignore-not-found
+    exit 0
+fi
+
+cm_enabled="$(tilt get configmap "$child_name" --ignore-not-found -o jsonpath='{.data.enabled}')"
+if [[ "$cm_enabled" == "" ]]; then
```

Comment:
IIUC we're only setting `ports` the first time the configmap gets created, and if an object has one link and then adds a second, we won't add the second to the configmap. This obviously isn't a blocker to the extension being useful, but might be worth a TODO.

Reply:
fixed!